\begin{enumerate}
\item Supplementary Figure~\ref{fig:workflow}
\item Supplementary Figure~\ref{fig:motclust}
\item Supplementary Figure~\ref{fig:fitness}
\item Supplementary Figure~\ref{fig:gresVsOperons}
\item Supplementary Figure~\ref{fig:corems}
\item Supplementary Figure~\ref{fig:suppfig6}
\end{enumerate}

\begin{figure}[!b]
\centering
\epsfig{file=figures/e1.eps,width=0.95\linewidth}
%\\\\\vspace{.07in}\hline\\\\\vspace{.07in}
\caption{{\bf Detailed workflow for EGRIN 2.0 inference
procedure.} Data input, processing and analysis to construct EGRIN 2.0
model for {\it H. salinarum} and {\it E. coli}, and predictions
generated. Predictions highlighted in individual figures are noted.}
\label{fig:workflow}
\vspace{-.1in}
\end{figure}

\begin{figure}[!b]
\centering
\epsfig{file=figures/e2.eps,width=0.8\linewidth}
%\\\\\vspace{.07in}\hline\\\\\vspace{.07in}
\caption{{\bf Genome-wide discovery and validation of gene
regulatory elements (GREs). (A) Motif clustering and GRE
identification.} (Left) A schematic of the approach used to align and
cluster individually detected motifs to define GREs. In this example,
similar motifs were aligned and clustered into three GREs using Tomtom
and mcl (Details in Methods and Supplementary Methods). (Center) The
{\it H. salinarum} network of aligned and clustered motifs. (Right)
Two {\it H. salinarum} GREs discovered by this method. The motif logo
of each GRE was generated by summing PSSMs of the individual aligned
motifs in the cluster, as illustrated by three examples of individual
motifs (prior to alignment) for each of the two GREs. Note that
relative to the individual motifs, the averaged GRE motif is more
palindromic - a hallmark of binding sites for dimeric TFs. {\bf (B)
Integration of GRE discovery and Inferelator predictions yields
comprehensive and detailed gene regulatory networks.} EGRIN
2.0-inferred {\it E. coli} regulatory subnetwork for two genes (green
circles) in the PurR/ArgR regulon: carA (b0032) and pyrL (b4246).  The
EGRIN 2.0 predictions are divided into GRE-based (dark violet) and
Inferelator-based (red), and compared to predictions (or annotations)
from other algorithms/databases (yellow: CLR; green: DREAM5 ensemble;
black: RegPrecise; blue: RegulonDB). In two cases
(ArgR$\rightarrow$carA and ArgR$\rightarrow$pyrL), EGRIN 2.0 discovers
regulatory interactions that were missed by either hand-curated
databases or expression-based inference procedures. {\bf (C) Number of TFs
in RegulonDB re-discovered by various methods.} Comparison of EGRIN 2.0
(solid line, far right) to individual \cm\ runs, as well as
multiple runs of k-means, WGCNA, and DISTILLER on subsets of the
expression data. Evaluation made with respect to re-discovery of 88
TFs in RegulonDB with $\geq 3$ unique sites based on genome-wide
binding site locations (FDR $\leq 0.05$). {\bf (D) Genome-wide
distribution of GREs relative to experimentally mapped transcriptional
start sites in {\it H. salinarum}.} (Left) Predicted positions for all GREs
in gene promoters upstream of experimentally mapped transcription
start sites (TSSs; Koide et al., 2009) in and (Right) four example
elements. Distribution peaks for most GREs occur at characteristic
locations. For instance, the location of TATA box-like elements
(GRE \#25) between -21 to -40 nt upstream to TSSs in {\it H. salinarum} is
consistent with the characterized location of basal elements in
archaeal promoters (-25 to 30 nt upstream to TSS). GRE location
enables prediction of putative roles for the cognate TF (\eg
repressor, activator or a basal factor).}
\label{fig:motclust}
\vspace{-.1in}
\end{figure}

\begin{figure}[!b]
\centering
\epsfig{file=figures/e3.eps,width=0.95\linewidth}
%\\\\\vspace{.07in}\hline\\\\\vspace{.07in}
\caption{{\bf EGRIN 2.0 models dynamic regulatory mechanisms that
    result in highly correlated fitness effects.}  
{\bf (A)} (Left) Enrichment for highly correlated, pairwise fitness
  measurements in gene knock outs across 324 conditions before and
  after removing gene associations annotated by operons
  (MicrobesOnline) and regulons (RegulonDB and RegPrecise) (KS-test,
  $D$-statistic). Two-thirds of gene-pairs with most highly correlated
  fitness within corems are not annotated by operons or
  regulons. (Right) Number of genes and associations predicted. {\bf (B)
  Deciphering GREs responsible for regulating corems.} A GRE is
  implicated in regulation of a corem when it is both (1) located
  within an expanded region (-875nt to +125nt) around the translation
  start site of any gene in the corem; and (2) present in biclusters
  containing a large fraction of corem genes (top decile). Relative
  GRE influence is computed as the frequency with which each GRE was
  discovered in these representative biclusters (see Supplementary
  Methods for more details). Influence scores are illustrated as pie
  charts and reported for each gene individually (\eg, VNG2347G); and
  as a composite by averaging across all genes in a corem. The width
  of each sector in the pie charts is proportional to the frequency of
  GRE discovery.  {\bf (C)} (Top) Predicted promoter architecture for
  {\it E. coli} pyrL (b4246). Overlapping GREs matching to PurR (GRE \#4) and
  ArgR (GRE \#12) were detected upstream of pyrL. These sites were not
  annotated in RegulonDB, but were validated in independent ChIP-chip
  experiments (Cho et al., 2012; Cho et al., 2011). Transcription
  start site indicated with arrow. (Bottom) Condition-specific
  promoter architectures for {\it E. coli} pyrL (as in Figure 2E). Variation
  in predicted GRE activity across three different subsets of
  experimental conditions (counts and fold-change) for two GREs in the
  pyrL promoter. Experimental subsets correspond to conditions under
  which at least one of three nucleotide biosynthetic corems is
  regulated (denoted by colored names at top-right of each plot) }
\label{fig:fitness}
\vspace{-.1in}
\end{figure}

\begin{figure}[!b]
\centering
\epsfig{file=figures/e4.eps,width=0.95\linewidth}
%\\\\\vspace{.07in}\hline\\\\\vspace{.07in}
\caption{{\bf GREs regulate internal modulation of operons in
    {\it H. salinarum}.}  {\bf (A) GREs located inside operons coincide with
  experimentally measured transcriptional break sites.} Experimentally
  determined transcription break sites (red dashed lines) above
  expression profiles of these regions across growth (heatmap, Koide
  et al., 2009) and ChIP-chip TFBs (Facciotti et al., 2007, vertical
  arrows) support the role of GREs in regulating segmentation of the
  operon in certain conditions. Insets contain regions immediately
  surrounding transcriptional break sites, including counts of GREs
  discovered at these locations. Three examples of conditionally
  segmented operons are shown. {\bf (B) Alternate regulatory modes for dpp
  operon predicted by corems.} Corems group together functionally
  related sets of genes that are co-regulated in similar environments
  by similar factors (Left) Presence/absence of dpp operon genes in
  corems. Three classes of corems exist for the dpp operon: (1) the
  entire operon (\eg hc21645), (2) the leader gene dppA
  (\eg hc6326), and (3) five ``tail'' genes excluding dppA
  (hc21279). (Middle) Gene similarity between corems (heatmap, Jaccard
  index). Functional annotations of genes in three highly similar
  clusters of corems to right. GRE composition for three corems shown
  below (pie chart, see Figure E3B). (Right) Similarity of conditions
  regulated (heatmap, upper triangle, Jaccard index) and GREs
  (heatmap, lower triangle, Jaccard index) among corems. Ordering is
  identical to (Middle). Environmental Ontology term enrichment (see
  Supplemental Methods) for three clusters depicted to right. {\bf (C) GRE
  compositions coincide with corem membership.} Network representation
  for three corems described above. Genes represented by circles. Edge
  colors and colored region behind the network indicate corem
  membership. Pie charts reflect GRE composition of each gene (see
  Figure E3B). Key for pie charts at top. Shading behind nodes (center
  of network) indicates dpp operon genes. {\bf (D) dppA is more tightly
  co-expressed with genes of hc6326 in some environments than the
  other genes in the dpp operon.} Relative expression of dppA compared
  to (left) other genes of dpp opeon and (right) hc6326. {\bf (E) EGRIN
  2.0 predicts conditional modulation of dpp operon in {\it E. coli} as
  well.} Promoter architecture within intergenic space between dppA and
  dppB suggested locations for TF binding internal to the operon (as
  in {\bf (A)}). GRE binding sites are proximal to an experimentally
  characterized IHF binding site (black horizontal bar; RegulonDB).  }
\label{fig:gresVsOperons}
\vspace{-.1in}
\end{figure}

\begin{figure}[!b]
\centering
\epsfig{file=figures/e5.eps,width=0.95\linewidth}
%\\\\\vspace{.07in}\hline\\\\\vspace{.07in}
\caption{{\bf 4. Corems integrate and subdivide regulons and operons
    in {\it E. coli} to coordinate related biological processes.}  {\bf (A)
    Corems accurately predict conditionally modulated operons in
    {\it E. coli}.} GREs coincide with experimentally measured break
  sites. Three examples of experimentally determined transcription
  break sites (red dashed lines) in operons predicted by corems to be
  conditionally segmented. Expression levels of these regions were
  profiled across growth in rich media (heatmap). Inset contains
  region immediately surrounding a transcriptional break site,
  including counts of GREs discovered at these locations (as in Figure
  E4A). {\bf (B) Corems model the mechanistic basis for conditional
    subdivision of the PurR regulon.} (Left) Corems identify the most
  highly correlated subgroupings of genes in PurR regulon. Gene
  expression correlation across all experiments (upper triangle)
  compared to similarity of corem membership (lower-triangle, Jaccard
  index) for genes of the PurR regulon (gene identifiers expanded to
  right). (Right) Similarity of regulated conditions (upper triangle,
  Jaccard index) and GREs composition for these genes (bottom
  triangle, Jaccard index). Consistent patterns of
  conditional-activity and GRE composition in their promoter regions
  further supports subdivision of PurR genes into separate
  corems. Gene order is same as left. {\bf (C) Corems reflect
    functional integration across diverse regulatory mechanisms.}
  Network representation for three of the corems described above (and
  in the text). Genes are represented by circles. Edge colors and
  colored region behind the network indicate corem membership. Pie
  charts reflect GRE composition of each gene (see
  Figure~\ref{fig:fitness}B). Key for pie charts at bottom. GRE-TF
  matches are indicated. Shading behind nodes denotes PurR regulon
  genes. At least 7 different mechanisms regulate the expression of
  these genes.  }
\label{fig:corems}
\vspace{-.1in}
\end{figure}

\begin{figure}[!b]
\centering
\epsfig{file=figures/e6.eps,width=0.95\linewidth}
%\\\\\vspace{.07in}\hline\\\\\vspace{.07in}
\caption{{\bf Organization of genes in corems provides meaningful link between transcriptional regulation, metabolic dynamics and fitness.} 
{\bf (A) Genes from corems related to nucleotide biosynthesis have highly
  similar fitness effects when they are deleted.} (Left) Violin plot
  shows distribution of all fitness correlations for genes in three
  nucleotide biosynthesis-associated corems compared to all genes in
  the data set. (Right) KS $D$-Statistic relates to enrichment for
  highly correlated gene-gene fitness associations in the corems. All
  three corems enrich for similar fitness effects (KS FDR $< 5\times 10^{-9}$).
  {\bf (B) Corems model fitness effects that occur in specific
  environments.} Violin plots show distribution of relative fitness
  among corems across conditions (negative values indicate lower
  fitness relative to WT). Brief condition descriptions are displayed
  below. Shading within the violin plot indicates that the
  distribution of fitness values is significantly in that condition
  (KS-test FDR $\leq 0.05$). Fitness values for the subset of genes from
  the PurR regulon that do not occur in ec516031 are displayed to the
  left. These genes do not have significant fitness effects in any of
  the environments tested. Data from (Nichols et al., 2011). {\bf (C)
  Metabolite correlation explains co-regulation within
  metabolically-linked corems.} (Left) Expression correlation for TFs
  associated with three corems described in the text
  (ec516031,ec512157,ec516034). (Right) Correlation allosteric
  regulators for these TFs. TF regulated by each biomolecule listed in
  parentheses (Novichkov et al., 2010). Red boxes indicate PurR-ArgR
  and their corresponding effector molecules. Data from (Ishii et al.,
  2007).
}
\label{fig:suppfig6}
\vspace{-.1in}
\end{figure}
