
\textcolor{red}{\tmem{\textbf{Model validation data was not used for model construction. }}}

\subsection{\halo}\label{halodata}

\subsubsection{Tiling array transcriptome measurements}

We generated {\it H. salinarum NRC-1} high-resolution (12 nt) tiling
array transcriptome measurements over 12 points along the growth curve
in rich media. These were analyzed and published in a separate study
\cite{Koide2009}. Locations of putative transcription breaks in these data were
identified in \cite{Koide2009} using multivariate recursive
partitioning, including signals from both relative changes in
expression along the growth curve, as well as raw RNA hybridization
signal. For more details, see \cite{Koide2009}.

\subsubsection{ChIP-chip transcription factor binding measurements for global regulators}

Global binding of eight general transcription factors (seven TFBs
[TFBa, TFBb, TFBc, TFBd, TFBe, TFBf, and TFBg] and one TBP [TbpB]) and
three specific TFs (Trh3, Trh4, and VNG1451C) in {\it H. salinarum}
were collected in our lab by ChIP-chip. A detailed protocol is
described in \cite{Facciotti2007}. Briefly, ChIP-enriched and
amplified DNA for eleven regulators was hybridized to a low-resolution
(500~nt resolution) custom PCR-product array spotted in-house. The
resulting intensities were analyzed using
{\tmsamp{MeDiChI}} \cite{Reiss2008} to obtain binding site locations
with an average precision of 50~nt. Local false discovery rates
(LFDRs) were quantified by simulation. For more details on the
ChIP-chip analysis methodology used in this work,
see \cite{Reiss2008}.

\subsubsection{\textit{kdp} promoter serial truncation measurements}

{\it H. salinarum NRC-1} \textit{kdpFABC} truncation data were
obtained from \cite{Kixmueller2011}. Briefly, the authors measured
relative induction of a transcriptional reporter after serial
truncation of the \textit{H. salinarum} R1 \textit{kdpFABC}
promoter. The authors measured $\beta$-Galactosidase activities from
truncated transcriptional fusions of the \textit{kdpFABC} promoter
to \textit{bgaH}. $\beta$-Galactosidase activities were measured in
triplicate from cultures grown in inducing (3 mM K$^{+}$) and
non-inducing (100 mM K$^{+}$) conditions. We obtained data
corresponding to Figure~4 of the paper, in which the authors quantify
the fractional $\beta$-Galactosidase activity (non-induced/induced)
among the serial truncations (private communication). We overlaid
motif predictions from \egrine~on this data set to reach our
conclusions.

\subsection{\eco}\label{ecodata}

\subsubsection{Tiling array transcriptome measurements}
\label{section:ecoarray}

We measured \eco\ tiling array transcriptome profiles at nine
different time points during growth in rich media (LB). Growth phases
spanned lag-phase (OD600 = 0.05) to late stationary-phase (OD600 =
7.3). RNA samples were prepared by hot phenol-chloroform
extraction \cite{Khodursky2003}. RNA was directly labeled and
hybridized to custom Agilent tiling arrays containing 60mer probes
tiled across both strands of the \eco\ genome using a sliding window
of 23~nt (GEO Platform GPL18392), as in \cite{Koide2009}. Expression
measurements were quantile-normalized as in \cite{Yoon2011} and
analyzed for condition-specific transcriptional isoforms following the
segmentation protocol described in \cite{Koide2009}. Data is available
on GEO (GSE55879).

\subsubsection{PurR/$\Delta$PurR expression data and ChIP-chip transcription factor binding sites} 

{\it E. coli} PurR/$\Delta$PurR expression data and ChIP-chip
transcription factor binding measurements collected in the presence of
adenine were taken from \cite{Cho2011a}. ChIP-chip relative
intensities were re-analyzed using
{\tmsamp{MeDiChI}} \cite{Reiss2008} to obtain binding site locations
with an average precision of $\sim$25~nt.

\subsubsection{Fitness measurements}
\label{section:fitness}

{\it E. coli} fitness measurements across 324 conditions were
generated by \cite{Nichols2011}. In short, the authors quantitated
growth rates for 3979 single gene deletions in each of 324
environments with variable stress, drug, and environmental
challenges. \textit{E. coli} mutant colony sizes were quantified on
agar plates.  Fitness correlations were obtained directly from the
authors: \href{http://ecoliwiki.net/tools/chemgen/}{http://ecoliwiki.net/tools/chemgen/}. Each
correlation value represents the Pearson correlation of fitness (\ie,
relative growth rate) for pairs of single gene deletion mutants
measured across all 324 conditions that are also present in our
analysis. Relative fitness scores were also obtained directly from the
authors.

\subsubsection{Effector molecule measurements}

{\it E. coli} effector molecule measurements were taken
from \cite{Ishii2007}. The authors measured metabolite levels using
capillary electrophoresis time-of-flight mass spectrometry (CE-TOFMS)
in \eco, as well as several other biomolecules (\eg., RNA and
protein). \textit{E. coli} was grown in a chemostat at several
different dilution rates (0.1, 0.2, 0.4, 0.5, and 0.7
hours$^{â€“1}$). We obtained the metabolite levels from the authors and
computed Pearson correlation between metabolites assigned to regulate
TFs by RegPrecise \cite{Novichkov2010}.

\subsubsection{Experimentally mapped {\it E. coli} transcription factor binding sites}

We compared genome-wide locations of GREs in the {\it E. coli} EGRIN
2.0 model with experimentally-mapped binding sites from the \rdb
database \cite{Gama-Castro2011}. To maintain consistency with our
comparisons against the DREAM5 community networks \cite{Marbach2012},
we used version 6.8 of the database. For binding sites, we used the
{\tmsamp{BindingSiteSet}} table, filtered for only interactions with
experimental evidence, and used only TFs with $\geq 3$ unique binding
sites -- a total of 88 TFs.

\subsubsection{Experimentally measured {\it E. coli} transcription factor regulatory targets}
\label{section:eco:gold:standard}

For the {\it E. coli} gold standard network, we used the same network
as that used by \cite{Marbach2012} for validation of the DREAM5 {\it
E. coli} community predicted regulatory networks. This gold standard
is based upon version 6.8 of the \rdb~
database \cite{Gama-Castro2011}, and only interactions with at least
one strong evidence were included, for a total of 2,066 interactions.
We mapped the $aaaX$-style gene names in the DREAM5 gold standard to
the $b1234$ in \cm~using a translation table compiled in the
{\tmsamp{EcoGene}} database, version 3.0 \cite{Zhou2013a}. We were
able to map a total of 4,273 gene names. The final gold standard
consisted of 2,064 interactions between 141 TFs and 997 target
genes. The final, complete gold standard network used for all analyses
is available \href{http://egrin2.systemsbiology.net/}{online}.
%at \ref{tables:DREAM5_gold_network.tsv}.
