
\DIFdelbegin \subsubsection{%DIFDELCMD < \halo%%%
}%DIFAUXCMD
\addtocounter{subsubsection}{-1}%DIFAUXCMD
\DIFdelend \DIFaddbegin \DIFadd{\textcolor{red}{\tmem{\textbf{Model validation data was not used for model construction. }}}
}

\subsection{\halo}\DIFaddend \label{halodata}

%DIF < \begin{enumerate}[(1)]
\DIFdelbegin \paragraph{\DIFdel{Tiling array transcriptome measurements}}
%DIFAUXCMD
\addtocounter{paragraph}{-1}%DIFAUXCMD
\paragraph{\DIFdel{ChIP-chip transcription factor binding measurements for global regulators}}
%DIFAUXCMD
\addtocounter{paragraph}{-1}%DIFAUXCMD
\paragraph{\DIFdel{Kdp promoter serial truncation measurements}}
%DIFAUXCMD
\addtocounter{paragraph}{-1}%DIFAUXCMD
%DIF < \end{enumerate}
\DIFdelend \DIFaddbegin \subsubsection{\DIFadd{Tiling array transcriptome measurements}}
\DIFaddend 

\DIFdelbegin \subsubsection{%DIFDELCMD < \eco%%%
}%DIFAUXCMD
\addtocounter{subsubsection}{-1}%DIFAUXCMD
%DIFDELCMD < \label{ecodata}
%DIFDELCMD < %%%
\DIFdelend \DIFaddbegin \DIFadd{We generated }{\it \DIFadd{H. salinarum NRC-1}} \DIFadd{high-resolution (12 nt) tiling
array transcriptome measurements over 12 points along the growth curve
in rich media. These were analyzed and published in a separate study
\mbox{%DIFAUXCMD
\cite{Koide2009}
}%DIFAUXCMD
. Locations of putative transcription breaks in these data were
identified in \mbox{%DIFAUXCMD
\cite{Koide2009}
}%DIFAUXCMD
using multivariate recursive
partitioning, including signals from both relative changes in
expression along the growth curve, as well as raw RNA hybridization
signal. For more details, see \mbox{%DIFAUXCMD
\cite{Koide2009}
}%DIFAUXCMD
.
}\DIFaddend 

\DIFdelbegin \paragraph{\DIFdel{Tiling array transcriptome measurements}}
%DIFAUXCMD
\addtocounter{paragraph}{-1}%DIFAUXCMD
\DIFdelend \DIFaddbegin \subsubsection{\DIFadd{ChIP-chip transcription factor binding measurements for global regulators}}
\DIFaddend 

\DIFaddbegin \DIFadd{Global binding of eight general transcription factors (seven TFBs
}[\DIFadd{TFBa, TFBb, TFBc, TFBd, TFBe, TFBf, and TFBg}] \DIFadd{and one TBP }[\DIFadd{TbpB}]\DIFadd{) and
three specific TFs (Trh3, Trh4, and VNG1451C) in }{\it \DIFadd{H. salinarum}}
\DIFadd{were collected in our lab by ChIP-chip. A detailed protocol is described
in \mbox{%DIFAUXCMD
\cite{Facciotti2007}
}%DIFAUXCMD
. Briefly, ChIP-enriched and amplified DNA for 
eleven regulators was hybridized to a low-resolution (500~nt resolution)
custom PCR-product array spotted in-house. The resulting intensities were
analyzed using }{\tmsamp{MeDiChI}} \DIFadd{\mbox{%DIFAUXCMD
\cite{Reiss2008}
}%DIFAUXCMD
to obtain binding
site locations with an average precision of 50~nt. Local false discovery rates (LFDRs) were quantified by simulation. For more details
on the ChIP-chip analysis methodology used in this work, see \mbox{%DIFAUXCMD
\cite{Reiss2008}
}%DIFAUXCMD
.
}

\subsubsection{\DIFadd{Kdp promoter serial truncation measurements}}

{\it \DIFadd{H. salinarum NRC-1}} \textit{\DIFadd{kdpFABC}} \DIFadd{truncation data were obtained from \mbox{%DIFAUXCMD
\cite{Kixmueller2011}
}%DIFAUXCMD
. Briefly, the authors measured relative induction of a transcriptional reporter after serial truncation of the }\textit{\DIFadd{H. salinarum}} \DIFadd{R1 }\textit{\DIFadd{kdpFABC}} \DIFadd{promoter. The authors measured $\beta$-Galactosidase activities from truncated transcriptional fusions of the }\textit{\DIFadd{kdpFABC}} \DIFadd{promoter to }\textit{\DIFadd{bgaH}}\DIFadd{. $\beta$-Galactosidase activities were measured in triplicate from cultures grown in inducing (3 mM K$^{+}$) and non-inducing (100 mM K$^{+}$) conditions. We obtained data corresponding to Figure~4 of the paper, in which the authors quantify the fractional $\beta$-Galactosidase activity (non-induced/induced) among the serial truncations (private communication). We overlaid motif predictions from }\egrine\DIFadd{~on this data set to reach our conclusions.  
}

\subsection{\eco}\label{ecodata}

\subsubsection{\DIFadd{Tiling array transcriptome measurements}}
\label{section:ecoarray}
\DIFaddend We measured \eco\DIFaddbegin \ \DIFaddend tiling array transcriptome profiles at nine different
time points during growth in \DIFdelbegin \DIFdel{LB, spanning }\DIFdelend \DIFaddbegin \DIFadd{rich media (LB). Growth phases spanned }\DIFaddend lag-phase (OD600 = 0.05) to
\DIFaddbegin \DIFadd{late }\DIFaddend stationary-phase (OD600 = 7.3). RNA samples were prepared \DIFdelbegin \DIFdel{as
in (Koide et al. , 2009). Tiling arrays (Agilent) were custom designed
with }\DIFdelend \DIFaddbegin \DIFadd{by hot phenol-chloroform extraction \mbox{%DIFAUXCMD
\cite{Khodursky2003}
}%DIFAUXCMD
. RNA was directly labeled and
hybridized to custom Agilent tiling arrays containing }\DIFaddend 60mer probes
tiled across both strands of the \eco\ genome using a sliding window
of \DIFdelbegin \DIFdel{23bp. Data }\DIFdelend \DIFaddbegin \DIFadd{23~nt (GEO Platform GPL18392), as in \mbox{%DIFAUXCMD
\cite{Koide2009}
}%DIFAUXCMD
. Expression measurements }\DIFaddend were quantile-normalized as in \DIFdelbegin \DIFdel{(Yoon et al., 2011) }\DIFdelend \DIFaddbegin \DIFadd{\mbox{%DIFAUXCMD
\cite{Yoon2011}
}%DIFAUXCMD
}\DIFaddend and analyzed for condition-specific transcriptional
isoforms \DIFdelbegin \DIFdel{as in (Koide et al.
}\DIFdelend \DIFaddbegin \DIFadd{following the segmentation protocol described
in \mbox{%DIFAUXCMD
\cite{Koide2009}
}%DIFAUXCMD
. Data is available on GEO (GSE55879).
}

\subsubsection{\DIFadd{PurR/$\Delta$PurR expression data and ChIP-chip transcription factor binding sites}} 

{\it \DIFadd{E. coli}} \DIFadd{PurR/$\Delta$PurR expression data and ChIP-chip
transcription factor binding measurements collected in the presence of
adenine were taken from \mbox{%DIFAUXCMD
\cite{Cho2011a}
}%DIFAUXCMD
. ChIP-chip relative
intensities were re-analyzed using
}{\tmsamp{MeDiChI}} \DIFadd{\mbox{%DIFAUXCMD
\cite{Reiss2008}
}%DIFAUXCMD
to obtain binding site locations
with an average precision of $\sim$25~nt.
}

\subsubsection{\DIFadd{Fitness measurements}}
\label{section:fitness}
{\it \DIFadd{E. coli}} \DIFadd{fitness measurements across 324 conditions were
generated by \mbox{%DIFAUXCMD
\cite{Nichols2011}
}%DIFAUXCMD
. In short, the authors quantitated
growth rates for 3979 single gene deletions in each of 324 environments 
with variable stress, drug, and environmental challenges. }\textit{\DIFadd{E. coli}} \DIFadd{mutant
colony sizes were quantified on agar plates.  
Fitness correlations were obtained directly from the authors: }\href{http://ecoliwiki.net/tools/chemgen/}{http://ecoliwiki.net/tools/chemgen/}\DIFadd{. Each correlation value represents
the Pearson correlation of fitness (}\ie\DIFadd{, relative growth rate) for
pairs of single gene deletion mutants measured across all 324
conditions that are also present in our analysis. Relative fitness scores were also obtained directly from the authors. 
}

\subsubsection{\DIFadd{Effector molecule measurements}}

{\it \DIFadd{E. coli}} \DIFadd{effector molecule measurements were taken from \mbox{%DIFAUXCMD
\cite{Ishii2007}
}%DIFAUXCMD
. The authors measured metabolite levels using capillary electrophoresis time-of-flight mass spectrometry (CE-TOFMS) in }\eco\DIFadd{, as well as several other biomolecules (}\eg\DIFadd{., RNA and protein). }\textit{\DIFadd{E. coli}} \DIFadd{was grown in a chemostat at several different dilution rates (0.1}\DIFaddend , \DIFdelbegin \DIFdel{2009) .
}\DIFdelend \DIFaddbegin \DIFadd{0.2, 0.4, 0.5, and 0.7 hours$^{â€“1}$). We obtained the metabolite levels from the authors and computed Pearson correlation between metabolites assigned to regulate TFs by RegPrecise \mbox{%DIFAUXCMD
\cite{Novichkov2010}
}%DIFAUXCMD
.
}\DIFaddend 

\DIFdelbegin \paragraph{\DIFdel{PurR/$\Delta$PurR expression data and ChIP-chip transcription factor binding measurements}} 
%DIFAUXCMD
\addtocounter{paragraph}{-1}%DIFAUXCMD
\paragraph{\DIFdel{Fitness measurements}}
%DIFAUXCMD
\addtocounter{paragraph}{-1}%DIFAUXCMD
\paragraph{\DIFdel{Effector molecule measurements}}
%DIFAUXCMD
\addtocounter{paragraph}{-1}%DIFAUXCMD
\paragraph{%DIFDELCMD < \rdb\  %%%
\DIFdel{database of experimentally mapped transcription factor targets}}
 %DIFAUXCMD
\addtocounter{paragraph}{-1}%DIFAUXCMD
\DIFdelend \DIFaddbegin \subsubsection{\DIFadd{Experimentally mapped }{\it \DIFadd{E. coli}} \DIFadd{transcription factor binding sites}}

\DIFadd{We compared genome-wide locations of GREs in the }{\it \DIFadd{E. coli}} \DIFadd{EGRIN
2.0 model with experimentally-mapped binding sites from the }\rdb
\DIFadd{database \mbox{%DIFAUXCMD
\cite{Gama-Castro2011}
}%DIFAUXCMD
. To maintain consistency with our
comparisons against the DREAM5 community networks \mbox{%DIFAUXCMD
\cite{Marbach2012}
}%DIFAUXCMD
,
we used version 6.8 of the database. For binding sites, we used the
}{\tmsamp{BindingSiteSet}} \DIFadd{table, filtered for only interactions with
experimental evidence, and used only TFs with $\geq 3$ unique binding
sites -- a total of 88 TFs.
}

\subsubsection{\DIFadd{Experimentally measured }{\it \DIFadd{E. coli}} \DIFadd{transcription factor regulatory targets}}
\label{section:eco:gold:standard}

\DIFadd{For the }{\it \DIFadd{E. coli}} \DIFadd{gold standard network, we used the same network
as that used by \mbox{%DIFAUXCMD
\cite{Marbach2012}
}%DIFAUXCMD
for validation of the DREAM5 }{\it
\DIFadd{E. coli}} \DIFadd{community predicted regulatory networks. This gold standard
is based upon version 6.8 of the }\rdb\DIFadd{~
database \mbox{%DIFAUXCMD
\cite{Gama-Castro2011}
}%DIFAUXCMD
, and only interactions with at least
one strong evidence were included, for a total of 2,066 interactions.
We mapped the $aaaX$-style gene names in the DREAM5 gold standard to
the $b1234$ in }\cm\DIFadd{~using a translation table compiled in the
}{\tmsamp{EcoGene}} \DIFadd{database, version 3.0 \mbox{%DIFAUXCMD
\cite{Zhou2013a}
}%DIFAUXCMD
. We were
able to map a total of 4,273 gene names. The final gold standard
consisted of 2,064 interactions between 141 TFs and 997 target
genes. The final, complete gold standard network used for all analyses
is available at \ref{tables:DREAM5_gold_network.tsv}.
 }\DIFaddend